src/main.ts

```diff
@@ -152,6 +152,19 @@ export async function activate(context: vscode.ExtensionContext) {
         }
     }))
 
+    context.subscriptions.push(vscode.workspace.createFileSystemWatcher('**/*.tex', true, false, true).onDidChange((e: vscode.Uri) => {
+        if (vscode.window.activeTextEditor && vscode.window.activeTextEditor.document.fileName === e.fsPath) {
+            return
+        }
+        const configuration = vscode.workspace.getConfiguration('latex-workshop')
+        if (!configuration.get('latex.autoBuild.enabled') || extension.builder.disableBuildAfterSave) {
+            return
+        }
+        if (extension.manager.isTex(e.fsPath)) {
+            extension.commander.build()
```

Comment:
Instead of calling `build()` in `commander` (which is designed to be invoked upon vscode commands), you may use the same code from https://github.com/James-Yu/LaTeX-Workshop/pull/150/files#diff-c7f5bd618dbb862973c3204edf8120d6R20 to build the file.